vaadin-card-flow-parent/vaadin-card-flow/src/main/java/com/vaadin/flow/component/card/Card.java

```diff
@@ -85,26 +88,96 @@ public Component getMedia() {
+     *            the title property
+     * @see #setTitle(String, Integer)
+     * @see #setTitleHeadingLevel(Integer)
+     * @see #getTitleAsText()
+     */
+    public void setTitle(String title) {
+        doSetTitle((Component) null);
+        doSetTitle(title);
+    }
+
+    /**
+     * Sets the title and the heading level for the title. If a
+     * {@link #setHeader(Component) header component} is set, the title will not
+     * be displayed. Setting a title this way removes any title component set
+     * using {@link #setTitle(Component)}. Setting {@code null} or empty title
+     * removes any previously set {@code String} titles.
+     *
+     * @param title
+     *            the title property
+     * @param titleHeadingLevel
+     *            the title heading level property,
+     * @see #setTitle(String)
+     * @see #setTitleHeadingLevel(Integer)
+     * @see #getTitleAsText()
+     */
+    public void setTitle(String title, Integer titleHeadingLevel) {
+        setTitleHeadingLevel(titleHeadingLevel);
+        setTitle(title);
+    }
+
+    /**
+     * Sets the title heading level property for the titles set using
+     * {@link #setTitle(String)}. The default is 2. Setting {@code null} resets
+     * it to default. Does not affect the title components set using
+     * {@link #setTitle(Component)}.
+     *
+     * @param titleHeadingLevel
+     *            the title heading level property, {@code null} to remove
+     */
+    public void setTitleHeadingLevel(Integer titleHeadingLevel) {
+        if (titleHeadingLevel == null) {
+            getElement().removeProperty(TITLE_HEADING_LEVEL_PROPERTY);
+        } else {
+            getElement().setProperty(TITLE_HEADING_LEVEL_PROPERTY,
+                    titleHeadingLevel);
+        }
+    }
+
     /**
      * Sets the component used as the card's title. If a
      * {@link #setHeader(Component) header component} is set, the title will not
-     * be displayed.
+     * be displayed. This also removes any previously set {@code String} titles.
      * <p>
      * Passing {@code null} removes the current title from the card.
      *
      * @param title
      *            the title component, or {@code null} to remove
      */
     public void setTitle(Component title) {
-        SlotUtils.setSlot(this, TITLE_SLOT_NAME, title);
+        doSetTitle((String) null);
+        doSetTitle(title);
+    }
+
+    /**
+     * Gets the value of the {@code cardTitle} property. Returns empty if no
+     * title is set.
+     *
+     * @return the value of the title property
+     */
+    public String getTitleAsText() {
+        return getElement().getProperty(CARD_TITLE_PROPERTY, "");
     }
 
     /**
-     * Gets the current title component.
+     * Gets the current title component set using {@link #setTitle(Component)}.
      *
      * @return the title component, or {@code null} if none is set
      */
     public Component getTitle() {
+        if (!getTitleAsText().isEmpty()) {
```

Comment:
Do we need this? Assuming the update logic is correct, there can not be anything in the slot when a string title is set.

Reply:
Removed as not necessary anymore.